source/common/config/delta_subscription_impl.h

```diff
@@ -0,0 +1,223 @@
+#pragma once
+
+#include <queue>
+
+#include "envoy/api/v2/discovery.pb.h"
+#include "envoy/common/token_bucket.h"
+#include "envoy/config/subscription.h"
+
+#include "common/common/assert.h"
+#include "common/common/backoff_strategy.h"
+#include "common/common/logger.h"
+#include "common/common/token_bucket_impl.h"
+#include "common/config/grpc_stream.h"
+#include "common/config/utility.h"
+#include "common/grpc/common.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Config {
+
+struct ResourceNameDiff {
+  std::vector<std::string> added_;
+  std::vector<std::string> removed_;
+};
+
+/**
+ * Manages the logic of a (non-aggregated) delta xDS subscription.
```

Comment:
Eventually this is likely to need to take on the same role as `GrpcMuxImpl`; we'll want ADS or some other way to handle stronger-than-eventual consistency with incremental xDS. How do you plan on making that convergence happen?

Reply:
Well, the incremental logic added in this PR is still close enough to the vanilla xDS code it's derived from that it should be doable by just imitating the mechanisms used in vanilla ADS. However, I am somewhat hoping that I can find a cleaner way, along the lines of #5270.

Reply:
TODO added; aggregation is already tracked as an item in #4991.